lib/json/encoder.ex

```diff
@@ -148,6 +148,10 @@ end
 defimpl JSON.Encoder, for: Any do
   @any_encode "[Elixir.Any]"
 
+  def encode(%{} = struct) do
+    JSON.Encoder.Helpers.dict_encode(Map.to_list(struct))
+  end
+
   def encode(_), do: JSON.Encoder.encode(@any_encode)
   def typeof(_), do: JSON.Encoder.typeof(@any_encode)
```

Comment:
Can you add an overloaded `typeof` for the structs too? Otherwise it won't match